docs: Replace SSD sizing tool with cluster tiers for distributed/microservices mode (#15820)
grafana/loki: docs/sources/setup/size/_index.md

<!-- vale Grafana.Quotes = NO -->
<!-- vale Grafana.Quotes = YES -->

This section is a guide to size base resource needs of a Loki cluster.

Based on the expected ingestion volume, Loki clusters can be categorised into three tiers. Recommendations below are based on p90 resource utilisations of the relevant components. Each tab represents a different tier.

> Please use this document as a rough guide to specify CPU and Memory requests in your deployment. This is only documented for [microservices/distributed](https://grafana.com/docs/loki/latest/get-started/deployment-modes/#microservices-mode) mode at this time.

Reviewer: ```suggestion
Please use this document as a rough guide to specify CPU and Memory requests in your deployment. This is only documented for [microservices/distributed](https://grafana.com/docs/loki/<LOKI_VERSION>/get-started/deployment-modes/#microservices-mode) mode at this time.
```

Query resource needs can greatly vary with usage patterns and correct configurations. General notes on Query Performance:
- The rule of thumb is to run as small and as many queriers as possible. Unoptimised queries can easily require 10x of the suggested querier resources below in all tiers. Running horizontal autoscaling will be most cost effective solution to meet the demand.
- Use this [blog post](https://grafana.com/blog/2023/12/28/the-concise-guide-to-loki-how-to-get-the-most-out-of-your-query-performance/) to adopt best practices for optimised query performance.

> - Parallel-querier and related components can be sized the same along with queriers for starters, depending on how much Loki rules are used.

Reviewer: ```suggestion
- Parallel-querier and related components can be sized the same along with queriers to start, depending on how much Loki rules are used.
```

> - Large Loki clusters benefits from disk based caching solution, memcached-extstore. Please see a detailed [blog post](https://grafana.com/blog/2023/08/23/how-we-scaled-grafana-cloud-logs-memcached-cluster-to-50tb-and-improved-reliability/) and read more on [memcached/nvm-caching here](https://memcached.org/blog/nvm-caching/).

Reviewer: ```suggestion
- Large Loki clusters benefit from a disk based caching solution, memcached-extstore. Please see the detailed [blog post](https://grafana.com/blog/2023/08/23/how-we-scaled-grafana-cloud-logs-memcached-cluster-to-50tb-and-improved-reliability/) and read more about [memcached/nvm-caching here](https://memcached.org/blog/nvm-caching/).
```

- If you’re running a cluster that handles less than 30TB/day (~1PB/month) ingestion, we do not recommend configuring memcached-extstore. The additional operational complexity does not justify the savings.

> These are the node types we suggest from various cloud providers. Please see the relevant specs on the provider documents.

Reviewer: ```suggestion
These are the node types we suggest from various cloud providers. Please see the relevant specifications in your provider documentation.
```

<!-- vale Grafana.Quotes = NO -->
<!-- vale Grafana.Quotes = YES -->

> This section is a guide to size base resource needs of a Loki cluster.

Reviewer 2: I can't select the right line as a comment but lines 2 - 4 will need to be removed so it appears in the docs
```
_build:
  list: false
noindex: true
```

Based on the expected ingestion volume, Loki clusters can be categorised into three tiers. Recommendations below are based on p90 resource utilisations of the relevant components. Each tab represents a different tier.

> Please use this document as a rough guide to specify CPU and Memory requests in your deployment. This is only documented for microservices/distributed mode at this time.

Reviewer 2: ```suggestion
Please use this document as a rough guide to specify CPU and Memory requests in your deployment. This is only documented for [microservices/distributed](https://grafana.com/docs/loki/<LOKI_VERSION>/get-started/deployment-modes/#microservices-mode) mode at this time.
```

{{< /tab-content >}}
{{< tab-content name="~1PB/month (30TB/day)" >}}
| Component        | CPU Request | Memory Request (Gi)| Base Replicas | Total CPU Req |Total Mem Req (Gi)|
|------------------|-------------|-------------------|----------------|----------------|-----------------|
| Ingester         | 4           | 8                 | 150            | 600            | 1200            |
| Distributor      | 2           | 1                 | 100            | 200            | 100             |
| Index gateway    | 1           | 4                 | 20             | 20             | 80              |
| Querier          | 1.5         | 3                 | 250            | 375            | 750             |
| Query-frontend   | 1           | 4                 | 16             | 16             | 64              |
| Query-scheduler  | 2           | 0.5               | 2              | 4              | 1               |
| Compactor        | 6           | 40                | 1 (Singleton)  | 6              | 40              |
{{< /tab-content >}}
{{< /tabs >}}   

  <blockquote v-if="help">
    <span v-else-if="help === 'node'">

>     Defines the node type of the Kubernetes cluster. Is a vendor or type

Reviewer 2: Just looking at how this renders:
https://deploy-preview-loki-15820-zb444pucvq-vp.a.run.app/docs/loki/latest/setup/size/

I wonder if we move this span up and change the wording
```
<span v-else-if="help === 'node'">
We currently recommend these nodes when running a production cluster of Loki. If a vendor, type or latest release is missing please let us know!
</span>
```

Reviewer: I just checked the preview, yeah, we need to do something about hiding line 76 and 77 in the generated output.

Reviewer: Actually, if we're not computing a value, do we need to include anything after line 71?  🤔

Author: +1 we can totally remove it.
